src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -68,17 +69,10 @@ def post(url, body, headers = {}, options = {})
     end
 
     response = self.class.client_post(request_options)
-    ruby_response(response)
-  end
-
-  def get(url, headers={}, options={})
-
-    request_options = create_common_request_options(url, headers, options)
-    response = self.class.client_get(request_options)
-    ruby_response(response)
+    Puppet::Server::HttpResponse.new(response, url)
```

Comment:
This url here is wrong because it doesn't include the query params yet. Do you think it makes sense to do the param processing outside of creating the `RequestOptions`, and passing that result here, or would it be more correct to round-trip though Java, and make sure that the URL we put in the response is version as Java understood it?

Something like `Puppet::Server::HttpResponse.new(response, URI(request_options.uri.to_s))` seems to do the right thing if we want to round-trip, but not sure if that's overkill, or even a bad idea.

Reply:
I pushed up a commit with what I suggested here, but still not sure if it's necessary to do something that convoluted.